pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -207,9 +217,19 @@ public void setClients(final List<Client> clients) {
     public void setClients(final Client... clients) {
         this.clients = Arrays.asList(clients);
     }
-
+    
+    /**
+     * Returns initialized clients.
+     * 
+     * @return A list of clients ready to be used.
+     */
     public List<Client> getClients() {
-        return clients;
+    	init();
+    	if (this.clients == null) {
```

Comment:
What's the benefit of this test? Versus the original `CommonHelper.assertNotNull("clients", this.clients);`...